src/app/attendance/components/AttendanceList.tsx

```diff
@@ -0,0 +1,164 @@
+"use client";
+
+import React, { useEffect } from "react";
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import { useToast } from "@/hooks/use-toast";
+import { Attendance, Member } from "@/types";
+import { UserStatusBadge } from "@/common/StatusBadge/UserStatusBadge";
+import { StandardPage } from "@/common/StandardPage/StandardPage";
+import { Plus } from "lucide-react";
+import MemberTableActionRender from "@/app/members/components/MemberActionRender";
+
+interface AttendanceProps {
+  attendance: {
+    data: Attendance[];
+    success: boolean;
+    message: string;
+  };
+  members: {
+    data: Member[];
+    success: boolean;
+    message: string;
+  };
+}
+
+export const AttendanceList = ({ attendance, members }: AttendanceProps) => {
+  const { toast } = useToast();
+  const { data: attendanceData, success } = attendance;
+  const { data: memberData } = members;
+
+  const getNameById = (id: number) => {
+    const member = memberData.find((member) => member.id === id);
+
+    return member ? member.name : "Unknown";
+  };
+
+  const getShiftByMemberId = (memberId: number) => {
+    const member = memberData.find((member) => member.id === memberId);
+
+    return member && member.shift ? member.shift : "Unknown Shift";
+  };
+
+  const columns: ColumnDef<Attendance>[] = [
+    {
+      accessorKey: "memberId",
+      header: "Name",
+      cell: ({ row }) => (
+        <div className="capitalize">
+          {getNameById(row.getValue("memberId"))}
```

Comment:
Maybe we don't to pass members as prop. I think we can map over that in the server component and get required result there. Wdyt?